pkg/netstore/netstore_test.go

```diff
@@ -97,20 +101,74 @@ func TestNetstoreNoRetrieval(t *testing.T) {
 	}
 }
 
+// TestRecovery verifies if the recovery hook is called if the chunk is not there in the network
+// and the targets are present
 func TestRecovery(t *testing.T) {
 	hookWasCalled := make(chan bool, 1)
+	payload := make([]byte, swarm.ChunkSize)
+	_, err := rand.Read(payload)
+	if err != nil {
+		t.Fatal(err)
+	}
+	c, err := content.NewChunk(payload)
+	if err != nil {
+		t.Fatal(err)
+	}
 	rec := &mockRecovery{
-		hookC: hookWasCalled,
+		hookC:     hookWasCalled,
+		sendChunk: c,
 	}
 
 	retrieve, _, nstore := newRetrievingNetstore(rec)
-	addr := swarm.MustParseHexAddress("deadbeef")
+
 	retrieve.failure = true
 	ctx := context.Background()
 	ctx = sctx.SetTargets(ctx, "be, cd")
 
-	_, err := nstore.Get(ctx, storage.ModeGetRequest, addr)
-	if err != nil && !errors.Is(err, netstore.ErrRecoveryAttempt) {
+	rcvdChunk, err := nstore.Get(ctx, storage.ModeGetRequest, c.Address())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	select {
+	case <-hookWasCalled:
+		break
+	case <-time.After(100 * time.Millisecond):
+		t.Fatal("recovery hook was not called")
+	}
+
+	if !c.Address().Equal(rcvdChunk.Address()) {
```

Comment:
It make sense because, the actual chunk is embedded in to a SOC in the recovery function and unwrapped in the Get function. so this test the unwrapping if nothing else.